pkg/xds/envoy/clusters/v3/outlier_detection_configurer.go

```diff
@@ -0,0 +1,87 @@
+		SplitExternalLocalOriginErrors: c.CircuitBreaker.Spec.GetConf().SplitExternalAndLocalErrors,
+	}
+	c.configureTotalErrorDetector(cluster.OutlierDetection)
+	c.configureGatewayErrorDetector(cluster.OutlierDetection)
+	c.configureLocalErrorDetector(cluster.OutlierDetection)
+	c.configureStandardDeviationDetector(cluster.OutlierDetection)
+	c.configureFailureDetector(cluster.OutlierDetection)
+	return nil
+}
+
+func (c *OutlierDetectionConfigurer) configureTotalErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if total := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetTotalErrors(); total != nil {
+		outlierDetection.Consecutive_5Xx = total.GetConsecutive()
+		outlierDetection.EnforcingConsecutive_5Xx = &wrappers.UInt32Value{Value: 100}
+	} else {
+		outlierDetection.EnforcingConsecutive_5Xx = &wrappers.UInt32Value{Value: 0}
+	}
+}
+
+func (c *OutlierDetectionConfigurer) configureGatewayErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if gateway := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetGatewayErrors(); gateway != nil {
+		outlierDetection.ConsecutiveGatewayFailure = gateway.GetConsecutive()
+		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 100}
+	} else {
+		outlierDetection.EnforcingConsecutiveGatewayFailure = &wrappers.UInt32Value{Value: 0}
+	}
+}
+
+func (c *OutlierDetectionConfigurer) configureLocalErrorDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if local := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetLocalErrors(); local != nil {
+		outlierDetection.ConsecutiveLocalOriginFailure = local.GetConsecutive()
+		outlierDetection.EnforcingConsecutiveLocalOriginFailure = &wrappers.UInt32Value{Value: 100}
+	} else {
+		outlierDetection.EnforcingConsecutiveLocalOriginFailure = &wrappers.UInt32Value{Value: 0}
+	}
+}
+
+func (c *OutlierDetectionConfigurer) configureStandardDeviationDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if stdev := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetStandardDeviation(); stdev != nil {
+		outlierDetection.SuccessRateRequestVolume = stdev.GetRequestVolume()
+		outlierDetection.SuccessRateMinimumHosts = stdev.GetMinimumHosts()
+		if factor := stdev.GetFactor(); factor != nil {
+			outlierDetection.SuccessRateStdevFactor = &wrappers.UInt32Value{Value: uint32(factor.GetValue() * 1000)}
+		}
+		outlierDetection.EnforcingSuccessRate = &wrappers.UInt32Value{Value: 100}
+		outlierDetection.EnforcingLocalOriginSuccessRate = &wrappers.UInt32Value{Value: 100} // takes effect only when SplitExternalLocalOriginErrors is true
+	} else {
+		outlierDetection.EnforcingSuccessRate = &wrappers.UInt32Value{Value: 0}
+	}
+}
+
+func (c *OutlierDetectionConfigurer) configureFailureDetector(outlierDetection *envoy_cluster.OutlierDetection) {
+	if failure := c.CircuitBreaker.Spec.GetConf().GetDetectors().GetFailure(); failure != nil {
```

Comment:
According to https://github.com/kumahq/kuma/blob/ff2e4f2eab0d5da3f08397692a4c69c2b9a9513d/api/mesh/v1alpha1/circuit_breaker.pb.go#L175-L180
`c.CircuitBreaker.Spec.GetConf().GetDetectors()` can return nil